File: LEETCODE/242.cpp
//242. Valid Anagram


//Method 1 (using unordered map)
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        int flag=0;
        unordered_map<char,int> m;
        
        if(s.length()!=t.length())
            return false;
        
        for(int i=0;i<s.length();i++)
        {
            m[s[i]]++;
            m[t[i]]--;
        }
    
        for(auto i: m)
            if(i.second)
                return 0;
        
        return 1;
    }
};


//Method 2 (using sorting)
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        sort(s.begin(),s.end());
        sort(t.begin(),t.end());
        
        if(s==t)
            return 1;
        
        return 0;
        
    }
};

//Method 3 (using sorting)
class Solution {
public:
    bool isAnagram(string s, string t) {
    
        sort(s.begin(),s.end());
        sort(t.begin(),t.end());
        
        return s==t;
        
    }
};

File: LEETCODE/LEETCODE-PYTHON/242.py
# 242. Valid Anagram
# https://leetcode.com/problems/valid-anagram/description/

# Method-1 using sorting
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)==len(t):
            x=sorted(s)
            y=sorted(t)
            for i in range(len(s)):
                if x[i]!=y[i]:
                    return False
        else:
            return False

        return True
        
# Method-2 using freq counter
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        x=[0]*26
        if len(s)!=len(t):
            return False

        for i in range(len(s)):
            x[ord(s[i])-ord('a')]+=1
            x[ord(t[i])-ord('a')]-=1

        for i in range(26):
            if x[i]!=0:
                return False
        
        return True
